src/main/java/io/cryostat/net/web/http/api/beta/graph/GraphModule.java

```diff
@@ -0,0 +1,721 @@
+                                                            builder.toDisk(
+                                                                    (Boolean)
+                                                                            settings.get("toDisk"));
+                                                }
+                                                if (settings.containsKey("maxAge")) {
+                                                    builder =
+                                                            builder.maxAge(
+                                                                    (Long) settings.get("maxAge"));
+                                                }
+                                                if (settings.containsKey("maxSize")) {
+                                                    builder =
+                                                            builder.maxSize(
+                                                                    (Long) settings.get("maxSize"));
+                                                }
+                                                IRecordingDescriptor desc =
+                                                        helper.startRecording(
+                                                                cd,
+                                                                builder.build(),
+                                                                (String) settings.get("template"),
+                                                                TemplateType.valueOf(
+                                                                        ((String)
+                                                                                        settings
+                                                                                                .get(
+                                                                                                        "templateType"))
+                                                                                .toUpperCase()));
+                                                return null;
+                                            },
+                                            true);
+                                } catch (Exception e) {
+                                    logger.error(e);
+                                    exceptions.add(e);
+                                }
+                            });
+            if (!exceptions.isEmpty()) {
+                throw new BatchedExceptions(exceptions);
+            }
+
+            return matches;
+        };
+    }
+
+    @Provides
+    @Singleton
+    @Named("recordings")
+    @SuppressFBWarnings(
+            value = "URF_UNREAD_FIELD",
+            justification =
+                    "The Recordings fields are serialized and returned to the client by the GraphQL engine")
+    static DataFetcher<Recordings> provideRecordingsFetcher(
```

Comment:
Let’s say I have several JVMs, some of which contain archived recordings. If I started a recording in all of my JVMs with a fixed duration, then fetched them after they’ve been automatically stopped and archived, how could I search for the recordings that I started in that specific batch operation?

Reply:
There would be nothing linking a recording (active or archived) to a specific batch operation that was used to create that recording. If we implemented something that allows us to store some more metadata about recordings (see https://github.com/cryostatio/cryostat/issues/709) then perhaps we could include some piece of information about the batch operation execution in that metadata, but I don't know what that would be - the exact timestamp of the operation's execution? Does the user need to provide an operation ID that we record in the metadata, or perhaps we generate such an ID to include in the metadata and we include the ID in the response we send to the user?

I would think that the useful kinds of query here would not be "show me the recordings that were created by this specific batch operation I performed" but "show me all recordings with start time greater than XYZ" or "show me all recordings from JVMs with annotation `com.app/suspiciousService`". The operation itself doesn't seem like it's the thing of interest - it's the recordings that are interesting, and the properties they have, or the properties of the JVMs that they come from.

Reply:
> it's the recordings that are interesting, and the properties they have, or the properties of the JVMs that they come from.

Ah I see, as long as we can select the JVMs we're interested in, we can get all of the recordings we need.

Reply:
Yea, exactly. There definitely appears to be some opportunity to give the client a way to filter out which recordings they are interested in, but that seems less necessary than filtering by target.